components/analytics/true_up_review.tsx

```diff
@@ -0,0 +1,230 @@
+    const license = useSelector(getLicense);
+    const isLicensed = license.IsLicensed === 'true';
+    const isStarter = getIsStarterLicense(license);
+    const licenseIsTrueUpEligible = isLicensed && !isCloud && !isStarter;
+    const telemetryEnabled = getConfig(getState()).EnableDiagnostics === 'true';
+    const trueUpReviewError = useSelector((state: GlobalState) => {
+        const errors = getSelfHostedErrors(state);
+        return Boolean(errors.trueUpReview);
+    });
+
+    useEffect(() => {
+        if (reviewStatus.getRequestState !== 'IDLE' || !licenseIsTrueUpEligible) {
+            return;
+        }
+
+        dispatch(getTrueUpReviewStatus());
+    }, [dispatch, reviewStatus.getRequestState, licenseIsTrueUpEligible]);
+
+    // Download the review profile as a base64 encoded json file when the review request is submitted.
+    useEffect(() => {
+        if (reviewProfile.getRequestState === 'LOADING') {
+            return;
+        }
+
+        if (reviewProfile.getRequestState === 'OK' && isAirGapped && !trueUpReviewError && reviewProfile.content.length > 0) {
+            // Create the bundle as a blob containing base64 encoded json data and assign it to a link element.
+            const blob = new Blob([reviewProfile.content], {type: 'application/text'});
+            const href = URL.createObjectURL(blob);
+
+            const link = document.createElement('a');
+            const date = moment().format('MM-DD-YYYY');
+            link.href = href;
+            link.download = `True Up-${license.Id}-${date}.txt`;
+            document.body.appendChild(link);
+            link.click();
+
+            // Remove link and revoke object url to avoid memory leaks.
+            document.body.removeChild(link);
+            URL.revokeObjectURL(href);
+        }
+    }, [isAirGapped, reviewProfile, reviewProfile.getRequestState, trueUpReviewError]);
+
+    const formattedDueDate = (): string => {
+        if (!reviewStatus.due_date) {
+            return '';
+        }
+
+        // Convert from milliseconds
+        const date = new Date(reviewStatus.due_date);
+        return moment(date).format('MMMM DD, YYYY');
+    };
+
+    const handleSubmitReview = () => {
+        dispatch(submitTrueUpReview());
+    };
+
+    const handleDownloadBundle = () => {
+        dispatch(submitTrueUpReview());
+    };
+
+    const dueDate = (
+        <div className='TrueUpReview__dueDate'>
+            <span>
+                <FormattedMessage
+                    id='admin.billing.trueUpReview.due_date'
+                    defaultMessage='Due '
+                />
+            </span>
+            <span>
+                {formattedDueDate()}
+            </span>
+        </div>
+    );
+
+    const submitButton = (
+        <button
+            className={classNames('btn btn-primary TrueUpReview__submit', {'TrueUpReview__submit--error': trueUpReviewError})}
+            onClick={isAirGapped ? handleDownloadBundle : handleSubmitReview}
```

Comment:
Is this condition required? It seems they both call the same action

Reply:
Definitely not. They did something different much earlier in this implementation, but no longer do.